tutorials/iot-device-to-device/index.md

```diff
@@ -0,0 +1,254 @@
+---
+title: Google Cloud IoT Core Device to Device Communication
+description: Learn how to relay messages from one device to another using Google Cloud Functions.
+author: gguuss
+tags: Google Cloud IoT Core, Cloud IoT, Google Cloud Functions
+date_published: 2017-12-05
+---
+
+Sometimes you want to have one device control another device. For example, lets
+say that you have a device (Device 1) that is a switch that reconfigures a
+second device (Device 2) to be on.  In this scenario, you would want for a
+change on the first device (switch 1 on) to trigger a change on the second device
+(device 2 on).
+
+One approach to triggering the configuration change is to:
+
+1. Send telemetry message from Device 1 to PubSub using Cloud IoT Core bridge
+1. Configure Google Cloud Function to receive message and send configuration change
+1. Receive configuration change on Device 2
+
+as demonstrated in the following diagram:
+
+![Overview of device to device communication](img/overview.png)
+
+This tutorial demonstrates how you can do each of these steps.
+
+## Create a Google Cloud IoT Core topic
+
+From the [Google Cloud Console](https://console.cloud.google.com/iot), create
+a Google Cloud IoT Core Device Registry. When you create it, either set the
+event notification topic to an existing PubSub topic or create a new topic to
+be used for the demo.
+
+## Upload the Google Cloud Function
+The dev2dev folder of the tutorial contains an example Google Cloud Function
+that contains the `getClient` and `setDeviceConfig functions from the
+[NodeJS manager sample](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/master/iot/manager).
+
```

Comment:
Consider formatting these steps as a list (it might require a little reworking, given how they're formatted now).